agimus_controller_ros/agimus_controller_ros/agimus_controller.py

```diff
@@ -151,36 +208,13 @@ def buffer_has_twice_horizon_points(self) -> bool:
         """
         return len(self.traj_buffer) >= 2 * self.params.ocp.horizon_size
 
-    def first_solve(self, x0: npt.NDArray[np.float64]) -> None:
-        """Initialize OCP, then do the first solve."""
-        # Initialize OCP
-        self.set_weighted_traj_attributes()
-        self.ocp = OCPCrocoHPP(self.rmodel, self.cmodel, self.params.ocp)
-        self.ocp.set_planning_variables(self.x_plan, self.a_plan)
-        self.ocp.set_weights_variables(
-            self.weight_q, self.weight_qdot, self.weight_pose
-        )
-
-        # First solve
-        self.mpc = MPC(self.ocp, self.x_plan, self.a_plan, self.rmodel, self.cmodel)
-        us_init = self.mpc.ocp.u_plan[: self.params.ocp.horizon_size - 1]
-        self.mpc.mpc_first_step(self.x_plan, us_init, x0)
-        self.next_node_idx = self.params.ocp.horizon_size
-        self.get_logger().info("First solve done ")
-
-    def solve(self, x0: npt.NDArray[np.float64]) -> None:
-        """get new trajectory point, reset and solve the OCP problem."""
-        if len(self.traj_buffer) > 0:
-            self.last_point = self.traj_buffer.popleft()
-        self.mpc.mpc_step(x0, self.last_point, self.effector_frame_name)
-
-    def send_control_msg(self, sensor_msg: Sensor) -> None:
+    def send_control_msg(self, ocp_res: OCPResults) -> None:
         """Get OCP control output and publish it."""
-        _, tau, K_ricatti = self.mpc.get_mpc_output()
+        # _, tau, K_ricatti = self.mpc.get_mpc_output()
         ctrl_msg = lfc_py_types.Control(
-            feedback_gain=K_ricatti,
-            feedforward=tau.reshape(self.rmodel.nv, 1),
-            initial_state=sensor_msg,
+            feedback_gain=ocp_res.ricatti_gains[0],
+            feedforward=ocp_res.feed_forward_terms[0].reshape(self.rmodel.nv, 1),
+            initial_state=sensor_msg_to_numpy(self.sensor_msg),
```

Comment:
Why not using the already converted msg from main?
[Line 199](https://github.com/agimus-project/agimus_controller/pull/152/files#diff-a3a0af0dc38989d5148ea6e8eaba833229c13b35f6ceb7623d3960ae00464f0bL199)